akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorRefFactoryAdapter.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed.internal.adapter
+import akka.actor.typed._
+import akka.annotation.InternalApi
+import akka.ConfigurationException
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[typed] object ActorRefFactoryAdapter {
+  def spawnAnonymous[T](
+      context: akka.actor.ActorRefFactory,
+      behavior: Behavior[T],
+      props: Props,
+      rethrowTypedFailure: Boolean): ActorRef[T] = {
+    try {
+      ActorRefAdapter(context.actorOf(internal.adapter.PropsAdapter(() => behavior, props, rethrowTypedFailure)))
+    } catch {
+      case ex: ConfigurationException if ex.getMessage.startsWith("configuration requested remote deployment") =>
+        throw new ConfigurationException("Remote deployment not allowed for typed actors", ex)
+    }
+  }
+
+  def spawn[T](
+      actorRefFactory: akka.actor.ActorRefFactory,
+      behavior: Behavior[T],
+      name: String,
+      props: Props,
+      rethrowTypedFailure: Boolean): ActorRef[T] = {
+    try {
+      ActorRefAdapter(
+        actorRefFactory.actorOf(
+          internal.adapter.PropsAdapter(() => Behavior.validateAsInitial(behavior), props, rethrowTypedFailure),
+          name))
+    } catch {
+      case ex: ConfigurationException if ex.getMessage.startsWith("configuration requested remote deployment") =>
+        throw new ConfigurationException("Remote deployment not allowed for typed actors", ex)
```

Comment:
Instead of repeating exception message perhaps private[akka] val them in an existing deployment-related object and call here? Because now if that message changes there are two places to maintain that, unless updated, would not catch it.

Reply:
:thum